Python/Product/Analysis/AnalysisHashSet.cs

```diff
@@ -505,6 +564,20 @@ struct Bucket {
             public int HashCode;        // the hash code of the contained key.
         }
 
+        struct BucketSet {
+            public BucketSet(int capacity, int count = 0) {
+                Buckets = capacity > 0 ? new Bucket[capacity] : Array.Empty<Bucket>();
+                Count = count;
+                if (count > capacity) {
+                    throw new ArgumentOutOfRangeException(nameof(count));
+                }
+            }
+
+            public readonly Bucket[] Buckets;
+            public int Count;
+            public int Capacity => Buckets?.Length ?? 0;
```

Comment:
`Capacity` can also be set in constructor

Reply:
This doesn't take up any storage space (`AnalysisHashSet` is our most frequently created type)